src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1104,6 +1104,34 @@ def get_node_public_ip_prefix_id(self) -> Union[str, None]:
         # this parameter does not need validation
         return node_public_ip_prefix_id
 
+    def get_enable_fips_image(self) -> bool:
+        """Obtain the value of enable_fips_image.
+
+        :return: bool
+        """
+        # read the original value passed by the command
+        raw_value = self.raw_param.get("enable_fips_image")
+        # try to read the property value corresponding to the parameter from the `mc` object
+        value_obtained_from_mc = None
+        if self.mc and self.mc.agent_pool_profiles:
+            agent_pool_profile = safe_list_get(
+                self.mc.agent_pool_profiles, 0, None
+            )
+            if agent_pool_profile:
+                value_obtained_from_mc = (
+                    agent_pool_profile.enable_fips
+                )
+
+        # set default value
+        if value_obtained_from_mc is not None:
+            enable_fips_image = value_obtained_from_mc
+        else:
+            enable_fips_image = raw_value
+
+        # this parameter does not need dynamic completion
+        # this parameter does not need validation
+        return enable_fips_image
```

Comment:
Would appreciate it if you could replace this with the concise version in [aks-preview](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/decorator.py#L202-L222).

Reply:
thanks for the comments. replace it with the concise version from aks-preview